Python/codecs.c

```diff
@@ -55,22 +66,33 @@ int
 PyCodec_Unregister(PyObject *search_function)
 {
     PyInterpreterState *interp = _PyInterpreterState_GET();
-    PyObject *codec_search_path = interp->codec_search_path;
-    /* Do nothing if codec_search_path is not created yet or was cleared. */
-    if (codec_search_path == NULL) {
+    /* Do nothing if codec data structures are not created yet. */
+    if (FT_ATOMIC_LOAD_INT_ACQUIRE(interp->codecs.initialized) == 0) {
```

Comment:
I think we should do this check under the `init_mutex` lock.

Reply:
What would that protect against? I should probably change it to `if (initialized != 1) return` rather than `if (initialized == 0) return`, but if this check needs to be done while holding the lock, [this one](https://github.com/python/cpython/pull/117530/files#diff-a22bcb6c26007e5597e92a6abb0a27ce987072fcfa7220623c869044c3097784R1491-R1493) should be too.

Reply:
The suggestion was more motivated about being easier to reason about than a correctness issue.

But I don't think it matters much because we don't actually lazily initialize codecs.